### DISCUSSION TITLE
Support for Javascript (langchainjs)

### DISCUSSION THREAD

**Discussion by henrikix** [NVIDIA-NeMo/Guardrails#38] [General]
Is there any chance you are going to also implement a javascript library that could be used in combination with [langchainjs](https://github.com/hwchase17/langchainjs) ?

If not then are you aware of any alternatives to what you are offering available in javascript?

Thanks

**Comment by rafa-9** [comment]
looking for same. Any JS equivalent? The only options I see are NeMo and this https://github.com/ShreyaR/guardrails which is also in python based.

**Comment by drazvan** [comment]
Unfortunately, we do not currently offer a javascript version of NeMo Guardrails and we are not aware of any package providing similar functionality. Depending on the use case, you can start a NeMo Guardrails server and make the calls from javascript. But if you're looking for a pure JS solution, making the calls directly from the browser/nodejs server, we don't have that yet. Hard to say if/when it will make it to the roadmap.

**Comment by abhaychebium** [comment]
https://github.com/NVIDIA/NeMo-Guardrails/issues/375
